lib/util/makeNoMethodSetStateRule.js

```diff
@@ -28,6 +28,15 @@ function makeNoMethodSetStateRule(methodName) {
     create: function(context) {
       const mode = context.options[0] || 'allow-in-func';
 
+      function nameMatches(name) {
+        let matches = name === methodName;
```

Comment:
This seems like it could return immediately if true, cleaning up the rest of the function